docs/source/array-chunks.rst

```diff
@@ -0,0 +1,244 @@
+Chunks
+======
+
+Dask arrays are composed of many Numpy arrays.  How these arrays are arranged
+can significantly affect performance.  For example for a square array you might
+arrange your chunks along rows, along columns, or in a more square-like
+fashion.  Different arrangements of Numpy arrays will be faster or slower for
+different algorithms.
+
+Thinking about and controlling chunking is important to optimize advanced
+algorithms.
+
+Specifying Chunk shapes
+-----------------------
+
+We always specify a ``chunks`` argument to tell dask.array how to break up the
+underlying array into chunks.  We can specify ``chunks`` in a variety of ways
+
+1.  A uniform dimension size like ``1000``, meaning chunks of size ``1000`` in each dimension
+2.  A uniform chunk shape like ``(1000, 2000, 3000)``, meaning chunks of size ``1000`` in the
+    first axis, ``2000`` in the second axis, and ``3000`` in the third.
+3.  Fully explicit sizes of all blocks along all dimensions,
+    like ``((1000, 1000, 500), (400, 400), (5, 5, 5, 5, 5))``
+4.  A dictionary specifying chunk size per dimension like ``{0: 1000, 1: 2000,
+    2: 3000}``.  This is just another way of writing the forms 2 and 3 above.
+
+Your chunks input will be normalized and stored in the third and most explicit
+form.  Note that ``chunks`` stands for "chunk shape" rather than "number of
+chunks" so specifying ``chunks=1`` means that you will have very many chunks,
+each with exactly one element.
+
+For performance, a good choice of ``chunks`` follows the following rules:
+
+1.  A chunk should be small enough to fit comfortably in memory.  We'll
+    have many chunks in memory at once.
+2.  A chunk must be large enough so that computations on that chunk take
+    significantly longer than the 1ms overhead per task that dask scheduling
+    incurs.  A task should take longer than 100ms.
```

Comment:
Isn't this a little extreme? Is 100ms/task suggested elsewhere in the docs? Also, the overhead is not true for the threaded scheduler.

Reply:
I'm just copying things around here.  This is not new content.  It's hard to give a single number here.  It varies strongly depending on the situation.  These are just rules of thumb.  I suggest that you hash this out with @jakirkham